src/org/traccar/database/DeviceManager.java

```diff
@@ -387,4 +402,54 @@ public void sendCommand(Command command) throws Exception {
         }
         return result;
     }
+
+    public void updateMotionState(long deviceId, Position position) {
```

Comment:
Yes, my idea was that you store any required state in device manager, but actual calculation code is here. This way we separate the code to make it easier to read.